lib/queue.js

```diff
@@ -1,38 +1,56 @@
 class Queue {
-    constructor() {
-        // this.store = ...
-        throw new Error("This method has not been implemented!");
-    }
+  constructor(size = 20) {
+    this.store = Array(size);
+    this.tail = 0;
+    this.head = 0;
+    this.length = size;
+  }
 
-    enqueue(element) {
-        throw new Error("This method has not been implemented!");
+  //     enqueue(value) - Adds the value to the back of the queue.
+  enqueue(element) {
+    if (this.store[this.tail] === undefined || this.store[this.tail] === null) {
+      this.store[this.tail] = element;
+      this.tail = (this.tail + 1) % this.length;
+    } else {
+      throw new Error("The buffer is full");
     }
+  }
+  //         dequeue - removes and returns a value from the front of the queue
 
-    dequeue() {
-        throw new Error("This method has not been implemented!");
-    }
+  dequeue() {
+    if (this.isEmpty()) return null;
+    const value = this.store[this.head];
+    this.store[this.head] = null;
+    this.head = (this.head + 1) % this.length;
+    return value;
+  }
 
-    front() {
-        throw new Error("This method has not been implemented!");
-    }
+  front() {
+    return this.store[this.head];
+  }
 
-    size() {
-        throw new Error("This method has not been implemented!");
-    }
+  size() {
+    return this.length;
+  }
```

Comment:
This should really return the number of elements in the Queue not the size of the underlying data structure.